controllers/handler/api.go

```diff
@@ -422,12 +417,21 @@ func (a *api) ingressForWebsocket() client.Object {
 			},
 			Labels: a.labels,
 		},
-		Spec: extensions.IngressSpec{
-			Backend: &extensions.IngressBackend{
-				ServiceName: APIName + "-websocket",
-				ServicePort: intstr.FromString("ws"),
+		Spec: createIngressSpec(APIName+"-websocket", "ws", intstr.FromString("ws").IntVal),
+	}
+	return ing
+}
+
+func createIngressSpec(serviceName, portName string, port int32) nwv1.IngressSpec {
+	return nwv1.IngressSpec{
+		DefaultBackend: &nwv1.IngressBackend{
+			Service: &nwv1.IngressServiceBackend{
+				Name: serviceName,
+				Port: nwv1.ServiceBackendPort{
+					Name:   portName,
+					Number: port,
+				},
 			},
 		},
 	}
-	return ing
 }
```

Comment:
Should I move this common function to `common.go`?

Reply:
Yes.